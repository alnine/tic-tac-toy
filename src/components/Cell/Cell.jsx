import React from 'react';
import PropTypes from 'prop-types';
import CellButton from './Cell.style';

const Cell = ({ value, onClick }) => {
  return <CellButton label={value} onClick={onClick} />;
};

export default Cell;

Cell.propTypes = {
  value: PropTypes.string,
};

Cell.defaultProps = {
  value: null,
};
