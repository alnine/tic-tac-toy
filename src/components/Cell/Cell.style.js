import styled from 'styled-components';
import { Button } from '../../common/Button';

const CellButton = styled(Button)`
  width: 2em;
  height: 2em;
  margin-top: -1px;
  margin-right: -1px;
  font-size: 1em;
  font-weight: bold;
  line-height: 2em;
  text-align: center;
  border: 1px solid #000000;
`;

export default CellButton;
