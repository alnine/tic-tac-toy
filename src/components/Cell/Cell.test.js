import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Cell from './Cell';

it('render correctly cell component', () => {
  const CellComponent = renderer.create(<Cell />).toJSON();
  expect(CellComponent).toMatchSnapshot();
});

it('render cell correctly with null value', () => {
  const CellComponent = mount(<Cell value={null} />);
  expect(CellComponent.prop('value')).toEqual(null);
});

it('check cell value', () => {
  const CellComponent = mount(<Cell value="X" />);
  expect(CellComponent.text()).toEqual('X');
});

it('check the onClick is working', () => {
  const onClick = jest.fn();
  const CellComponent = mount(<Cell onClick={onClick} />);
  CellComponent.simulate('click');
  expect(onClick).toBeCalledTimes(1);
});
