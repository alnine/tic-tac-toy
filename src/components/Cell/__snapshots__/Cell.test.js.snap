// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly cell component 1`] = `
<button
  className="sc-AxjAm kYzbGw sc-AxirZ eqGkKr"
  disabled={false}
  onClick={null}
/>
`;
